### PR TITLE
Switch to Microsoft.DotNet.XUnitExtensions

### PR DESCRIPTION
**Pull request by rainersigwald**
Let's delete a bunch of code from our repo and replace it with the modern version from Arcade.

We can't delete our assembly entirely because Arcade doesn't have the Fixture stuff that we depend on fairly heavily.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -32,6 +32,7 @@
     <!-- DotNetCliVersion MUST match the dotnet version in global.json.
          Otherwise, this version of dotnet will not be installed and the build will error out. -->
     <DotNetCliVersion>$([System.Text.RegularExpressions.Regex]::Match($([System.IO.File]::ReadAllText('$(MSBuildThisFileDirectory)..\global.json')), '"dotnet": "([^"]*)"').Groups.get_Item(1))</DotNetCliVersion>
+    <MicrosoftDotNetXUnitExtensionsVersion>6.0.0-beta.21329.2</MicrosoftDotNetXUnitExtensionsVersion>
```
Comment (Forgind) [inline eng/Versions.props:35]: Will this be auto-updated by darc?

Also, is there some way we can specify the version once and have it appear both here and in Version.Details.xml?

**Review by Forgind — APPROVED** [review]
Understanding check:
We'd previously copied a lot of code from Arcade. Here, you're deleting that code and adding a reference to Arcade so don't have to carry it around and get updates for (almost) free.

If someone had built our repo just before this, then pulled in this change, did not build, went offline, and tried to build again, that would fail, correct? I don't think we should care—just trying to think of edge cases.

File: eng/Versions.props (hunk 1 above)
Comment (rainersigwald) [reply eng/Versions.props:35]: > Will this be auto-updated by darc?

Yes, I expect this to be updated by our existing arcade->msbuild darc subscription.

> Also, is there some way we can specify the version once and have it appear both here and in Version.Details.xml?

I am not aware of one. Darc's design requires having it in two places updated by tooling.

**Comment by rainersigwald** [comment]
> We'd previously copied a lot of code from Arcade. Here, you're deleting that code and adding a reference to Arcade so don't have to carry it around and get updates for (almost) free.

Correct.

> If someone had built our repo just before this, then pulled in this change, did not build, went offline, and tried to build again, that would fail, correct? I don't think we should care—just trying to think of edge cases.

Correct (as with any package addition or version change).

**Pull request merged**